lxd/container_lxc.go

```diff
@@ -6883,13 +6887,155 @@ func (c *containerLXC) insertProxyDevice(devName string, m types.Device) error {
 	return nil
 }
 
+func (c *containerLXC) doNat(proxy string, device types.Device) error {
+	// Check NAT requirements
+	if device["bind"] != "host" {
+		return errors.New("NAT requirements not met")
+	}
+
+	listenAddr, err := parseAddr(device["listen"])
+	if err != nil {
+		return err
+	}
+
+	connectAddr, err := parseAddr(device["connect"])
+	if err != nil {
+		return err
+	}
+
+	address, _, err := net.SplitHostPort(connectAddr.addr[0])
+	if err != nil {
+		return err
+	}
+
+	if len(connectAddr.addr) > len(listenAddr.addr) {
+		// Cannot support single port -> multiple port
+		return errors.New("Cannot map single port to multiple ports")
```

Comment:
This is true in general though so why are we only checking this now in the NAT codepath?
Feels like this should go in the config validator instead.

Reply:
Also, we should be using fmt.Errorf instead of errors.New when returning an error that's not wrapped